CONTRIBUTING.md

```diff
@@ -0,0 +1,219 @@
+
+To file non-security issues:
+
+1. Click the **Issues** tab in the github repository,
+
+2. Click the **New Issue** button,
+
+3. Fill out all sections in the issue template and submit.
+
+## Contributing site changes, docs, and examples
+
+## Contributing code
+
+### Accepting Contributor License Agreement (CLA)
+
+Before you submit your first contribution please accept our Contributor License Agreement (CLA) here. When you send your first Pull Request (PR), GitHub will ask you to accept the CLA.
+
+There is no need to do this before you send your first PR.
+
+Subsequent PRs will not require CLA acceptance.
+
+If for some (unlikely) reason at any time CLA changes, you will get presented with the new CLA text on your first PR after the change.
+
+### Obtaining the Source Code and Building the Project
+
+#### Build Prerequisites
+
+**Prerequisite 1** - Install a Java Development Kit (JDK) version 8
+
+Building Ballerina requires a Java SE Development Kit (JDK) version 8 to be installed. You can download one from one of the following locations:
+
+* [http://www.oracle.com/technetwork/java/javase/downloads/index.html](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+* [http://openjdk.java.net/install/index.html](http://openjdk.java.net/install/index.html)
+
+> NOTE: Set an environment variable `JAVA_HOME` to the path name of the directory into which you installed the JDK release.
+
+**Prerequisite 2** - Install Maven 3.5.0 or later 
+
+Install Apache Maven 3.5.0 or a later version: [https://maven.apache.org/install.html](https://maven.apache.org/install.html)
+
+Set the `MAVEN_OPTS` environment variable to avoid the `Maven OutOfMemoryError`.
+
+```
+MAVEN_OPTS="-Xmx2048M -XX:MaxPermSize=1024m" 
+```
+
+Make sure you have an active Internet connection to download dependencies while building.
+
+**Prerequisite 3** - Node (v8.9.x or latest LTS release) + npm (v5.6.0 or later)
+
+**Prerequisite 4** - Docker 
+
+#### Building from the source
+
+Clone this repository using the following command.
+
+```
+git clone --recursive https://github.com/ballerinalang/ballerina
+```
+
+If you download the sources, you need to update the git submodules using the following command.
+
+```
+git submodule update --init 
+```
+
+Run the Maven command `mvn clean install` from the Ballerina root directory:
+
+Command | Description 
+--- | ---
+`mvn clean install` | Build and install the artifacts into the local repository.
+`mvn clean install -Dmaven.test.skip=true` | Build and install the artifacts into the local repository, without running any of the unit tests.
+`mvn clean install -P ballerina` | The `ballerina` profile is used to build only the modules necessary for the runtime distribution (i.e., excluding the tools, etc.)
+
+Extract the Ballerina distribution created at `distribution/zip/ballerina/target/ballerina-<version>-SNAPSHOT.zip`.

```

Comment:
Need to be explicit about this. `zip/ballerina` contains the runtime only. `zip/ballerina-tools/` contains the runtime + tools (e.g., Composer)